system_tests/multi_writer_fallback_test.go

```diff
@@ -1027,3 +1027,245 @@ func TestMultiWriterFallback_AnyTrustToCalldataOnBackendFailure(t *testing.T) {
+
+	// Deploy ReferenceDA validator contract
+	builder.WithReferenceDAContractsOnly()
+
+	builder.BuildL1(t)
+
+	// 2. Setup CustomDA provider server with control handles
+	// Initial max size: 10KB
+	initialMaxSize := 10_000
+	l1info := builder.L1Info
+	dataSigner := signature.DataSignerFromPrivateKey(l1info.GetInfoWithPrivKey("Sequencer").PrivateKey)
+	validatorAddr := l1info.GetAddress("ReferenceDAProofValidator")
+	customDAServer, customDAURL, writerControl := createReferenceDAProviderServerWithControl(t, ctx, builder.L1.Client, validatorAddr, dataSigner, 0, initialMaxSize)
+	defer func() {
+		if err := customDAServer.Shutdown(ctx); err != nil {
+			t.Logf("Error shutting down CustomDA server: %v", err)
+		}
+	}()
+
+	t.Logf("CustomDA server with control running at: %s (initial max size: %d)", customDAURL, initialMaxSize)
+
+	// 3. Configure sequencer node with CustomDA only
+	builder.nodeConfig.DA.ExternalProvider.Enable = true
+	builder.nodeConfig.DA.ExternalProvider.RPC.URL = customDAURL
+	builder.nodeConfig.DA.ExternalProvider.WithWriter = true
+
+	// Disable AnyTrust
+	builder.nodeConfig.DA.AnyTrust.Enable = false
+
+	// Disable fallback to on-chain to ensure we're testing resize, not fallback
+	builder.nodeConfig.BatchPoster.DisableDapFallbackStoreDataOnChain = true
+
+	// Configure batch posting to use size-based triggers
+	builder.nodeConfig.BatchPoster.MaxDelay = 60 * time.Second
+
+	// 4. Build L2
+	builder.L2Info = NewArbTestInfo(t, builder.chainConfig.ChainID)
+	builder.L2Info.GenerateAccount("User2")
+	cleanup := builder.BuildL2OnL1(t)
+	defer cleanup()
+
+	// 5. Setup follower node with same DA config
+	nodeConfigB := arbnode.ConfigDefaultL1NonSequencerTest()
+	nodeConfigB.BlockValidator.Enable = false
+
+	// CustomDA config
+	nodeConfigB.DA.ExternalProvider.Enable = true
+	nodeConfigB.DA.ExternalProvider.RPC.URL = customDAURL
+
+	// Disable AnyTrust
+	nodeConfigB.DA.AnyTrust.Enable = false
+
+	nodeBParams := SecondNodeParams{
+		nodeConfig: nodeConfigB,
+		initData:   &builder.L2Info.ArbInitData,
+	}
+	l2B, cleanupB := builder.Build2ndNode(t, &nodeBParams)
+	defer cleanupB()
+
+	// Phase 1: Build and post a large batch with initial max size (10KB)
+	t.Log("Phase 1: Generating transactions to create a ~10KB batch")
+
+	l1BlockBeforePhase1, err := builder.L1.Client.BlockNumber(ctx)
+	Require(t, err)
+
+	// Generate enough transactions to create a ~10KB batch
+	for i := 0; i < 250; i++ {
+		tx := builder.L2Info.PrepareTx("Owner", "User2",
+			builder.L2Info.TransferGas, big.NewInt(1e12), nil)
+		err := builder.L2.Client.SendTransaction(ctx, tx)
+		Require(t, err)
+	}
+
+	t.Log("Phase 1: Generated 250 transactions")
+
+	// Create L1 blocks to trigger batch posting
+	for i := 0; i < 30; i++ {
+		SendWaitTestTransactions(t, ctx, builder.L1.Client, []*types.Transaction{
+			builder.L1Info.PrepareTx("Faucet", "User", 30000, big.NewInt(1e12), nil),
+		})
+	}
+
+	// Wait for batch to post
+	time.Sleep(time.Second * 2)
```

Comment:
Comment 1: we can use `AdvanceL1` utility instead of a loop here
Comment 2 (covering also the follower node syncing): could we reuse `checkBatchPosting` method here? I think it would be beneficial for us to have the batch posting check as standardized as possible. If I'm not mistaken, currently `checkBatchPosting` relies on `MaxDelay=0`, but maybe we can force posting with `builder.L2.ConsensusNode.BatchPoster.MaybePostSequencerBatch(ctx)` and some other config flag?